### PR TITLE
Update to angular 4 including test, docs, and demo.

### PR DESCRIPTION
**Pull request by blackbaud-joshgerdes**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,31 +1,34 @@
...
+    "build": "npm run clean:srctemp && node config/utils/stage-ts.js && cd .srctemp && ../node_modules/.bin/ngc && cd ../ && npm run compile:sass && cd config/rollup && ../../node_modules/.bin/rollup -c rollup.config.js && ../../node_modules/.bin/rollup -c rollup.min.config.js && npm run clean:srctemp && cd ../.. && cp -r ./src/scss ./dist/scss",
     "clean": "rimraf -- dist",
-    "clean:full": "npm cache clean && rimraf -- node_modules doc coverage dist .srctemp webdriver-screenshotslocal webdriver-screenshotslocal-diffs",
+    "clean:full": "npm cache clean && rimraf -- node_modules doc coverage dist .srctemp webdriver-screenshotslocal webdriver-screenshotslocal-diffs skyux-spa-visual-tests/node_modules/ skyux-spa-visual-tests/dist/",
     "clean:releases": "rimraf -- releases",
     "clean:srctemp": "rimraf -- .srctemp",
-    "compile:sass": "node-sass src/scss/sky-contrib.scss dist/css/sky.css",
-    "lint": "tslint 'src/**/*.ts'",
+    "compile:sass": "./node_modules/node-sass/bin/node-sass src/scss/sky.scss dist/css/sky.css",
```
Comment (blackbaud-brandonstirnaman) [inline package.json]: This should be using just node-sass still (let NPM locate it, rather than direct path)

File: src/vendor.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ import 'ng2-dragula/ng2-dragula';
 import 'moment';
 
 // microedge rxstate
-import 'microedge-rxstate';
+import 'microedge-rxstate/dist';
```
Comment (blackbaud-brandonstirnaman) [inline src/vendor.ts]: Did this work without the `/dist` before? i.e. will other things doing this import now break with later versions of rxstate?

File: package.json (hunk 1 above)
Comment (blackbaud-joshgerdes) [inline package.json]: this is a copy over from the skyux project to keep things in sync

File: src/vendor.ts (hunk 2 above)
Comment (blackbaud-joshgerdes) [inline src/vendor.ts]: It did but with the upgrade of modules to run the latest skyux then it started being a difficult (and needing the `/dist`).

Comment (blackbaud-brandonstirnaman) [inline src/vendor.ts]: That doesn't make sense to me at all

Comment (blackbaud-joshgerdes) [inline src/vendor.ts]: as well it shouldn't, the packaging for rxstate is off, have a fix coming.

**Pull request merged**